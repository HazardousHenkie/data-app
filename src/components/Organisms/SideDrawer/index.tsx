import React, { useState } from 'react'

import Drawer, { DrawerProps } from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import MenuIcon from '@material-ui/icons/Menu'

const SideDrawer: React.FC<Omit<DrawerProps, 'anchor' | 'onClose'>> = ({
    children
}) => {
    const [openDrawer, setOpenDrawer] = useState(false)

    const toggleDrawer = (open: boolean) => (
        event: React.KeyboardEvent | React.MouseEvent
    ) => {
        if (
            event.type === 'keydown' &&
            ((event as React.KeyboardEvent).key === 'Tab' ||
                (event as React.KeyboardEvent).key === 'Shift')
        ) {
            return
        }

        setOpenDrawer(open)
    }

    return (
        <>
            <IconButton
                color="inherit"
                aria-label="open drawer"
                onClick={toggleDrawer(true)}
                edge="start"
            >
                <MenuIcon />
            </IconButton>

            <Drawer
                anchor="left"
                open={openDrawer}
                onClose={toggleDrawer(false)}
            >
                {children}
            </Drawer>
        </>
    )
}

export default SideDrawer
