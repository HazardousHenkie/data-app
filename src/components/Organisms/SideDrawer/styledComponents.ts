import styled from 'styled-components'

import Drawer from '@material-ui/core/Drawer'

const StyledDrawer = styled(Drawer)`
    &&& {
        z-index: 1400 !important;
    }
`

export default StyledDrawer
