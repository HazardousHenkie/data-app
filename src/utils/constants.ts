enum ENDPOINTS {
    LOGOUT = '/.netlify/functions/googleLogout',
    lOGIN = '/.netlify/functions/googleLogin',
    REFRESH_TOKEN = '/.netlify/functions/refreshToken',
    COUNTRIES = '/.netlify/functions/countries',
    FAVORITE_COUNTRIES = '/.netlify/functions/getFavoritedCountries',
    SAVE_COUNTRY = '/.netlify/functions/saveCountry/',
    DELETE_COUNTRY = '/.netlify/functions/deleteCountry/'
}

export default ENDPOINTS
