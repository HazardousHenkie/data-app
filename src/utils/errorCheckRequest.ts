import { put } from 'redux-saga/effects'
import { getRefreshTokenRequest } from 'globals/authentication/refreshToken/actions'
import { logoutRequest } from 'globals/authentication/logout/actions'

import { setError } from 'globals/globalErrors/actions'
import { ResponseError } from './request'

// test this one?
function* requestErrorCheck(error: ResponseError) {
    try {
        if (localStorage.getItem('userId')) {
            yield put(
                getRefreshTokenRequest(localStorage.getItem('userId') as string)
            )
        } else {
            throw Error('No userId in local storage.')
        }
    } catch {
        yield put(logoutRequest())
        yield put(setError(error))
    }
}
export default requestErrorCheck
