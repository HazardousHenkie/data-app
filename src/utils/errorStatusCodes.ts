enum ERROR_STATUS_CODES {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    SERVER_ERROR = 500,
    SERVICE_UNAVAILABLE = 503,
    NOT_FOUND = 404
}

export default ERROR_STATUS_CODES
