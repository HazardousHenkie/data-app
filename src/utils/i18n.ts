import i18n from 'i18next'
import Backend from 'i18next-xhr-backend'
import LanguageDetector from 'i18next-browser-languagedetector'
import { initReactI18next } from 'react-i18next'

export const appLocales = ['en', 'ja', 'nl']

i18n.use(Backend)
    .use(Backend)
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
        debug: false,
        lng: 'en',
        fallbackLng: 'en',
        whitelist: appLocales,
        interpolation: {
            escapeValue: false
        },
        react: {
            wait: true
        }
    })

export default i18n
