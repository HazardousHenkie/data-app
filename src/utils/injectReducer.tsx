import * as React from 'react'
import hoistNonReactStatics from 'hoist-non-react-statics'
import { useStore, ReactReduxContext } from 'react-redux'

import { getInjectors } from './reducerInjectors'
import { InjectReducerParams, InjectedStore } from 'types'

export default function hocWithReducer<P>({
  key,
  reducer
}: InjectReducerParams) {
  function wrap(
    WrappedComponent: React.ComponentType<P>
  ): React.ComponentType<P> {
    class ReducerInjector extends React.Component<P> {
      public static contextType = ReactReduxContext
      public static WrappedComponent = WrappedComponent
      public static displayName = `withReducer(${WrappedComponent.displayName ||
        WrappedComponent.name ||
        'Component'})`

      constructor(props: any, context: any) {
        super(props, context)

        getInjectors(context.store).injectReducer(key, reducer)
      }

      public render() {
        return <WrappedComponent {...this.props} />
      }
    }

    return hoistNonReactStatics(ReducerInjector, WrappedComponent) as any
  }
  return wrap
}

const useInjectReducer = ({ key, reducer }: InjectReducerParams) => {
  const store = useStore() as InjectedStore
  React.useEffect(() => {
    getInjectors(store).injectReducer(key, reducer)
  }, [key, reducer, store])
}

export { useInjectReducer }
