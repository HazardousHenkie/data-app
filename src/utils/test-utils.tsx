import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'configureStore'

import history from 'utils/history'
import { Provider } from 'react-redux'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState, history),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            {children}
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )
    }
    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions
    })
}

export * from '@testing-library/react'

export { customRender as render }
