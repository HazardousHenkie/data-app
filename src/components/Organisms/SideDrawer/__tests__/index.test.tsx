import React from 'react'
import { render, fireEvent } from 'utils/test-utils'

import StyledDrawer, { DrawerChildren } from '../styledComponents'

import SideDrawer from '../index'

describe('<SideDrawer />', () => {
    it('should render like snapshot', () => {
        const component = render(<SideDrawer />)

        expect(component).toMatchSnapshot()
    })

    it('should render IconButtonStyled', () => {
        const { getByTestId } = render(<SideDrawer />)

        const IconButtonStyled = getByTestId('IconButtonStyled')

        expect(IconButtonStyled).toBeInTheDocument()
    })

    it('should render aria-label on IconButtonStyled', () => {
        const { getByLabelText } = render(<SideDrawer />)

        const listLabel = getByLabelText('open drawer')

        expect(listLabel).toBeInTheDocument()
    })

    it('should render StyledDrawer when IconButtonStyled is clicked', () => {
        const { getByTestId } = render(<SideDrawer />)

        const IconButtonStyled = getByTestId('IconButtonStyled')
        fireEvent.click(IconButtonStyled)
        const StyledDrawerButton = getByTestId('StyledDrawer')

        expect(StyledDrawerButton).toBeInTheDocument()
    })

    it('It shouldn render StyledDrawer DrawerChildren', () => {
        const { getByText, getByTestId } = render(
            <SideDrawer>
                <StyledDrawer anchor="left" open>
                    <DrawerChildren>children</DrawerChildren>
                </StyledDrawer>
            </SideDrawer>
        )

        const IconButtonStyled = getByTestId('IconButtonStyled')
        fireEvent.click(IconButtonStyled)

        expect(getByText('children')).toBeInTheDocument()
    })
})
