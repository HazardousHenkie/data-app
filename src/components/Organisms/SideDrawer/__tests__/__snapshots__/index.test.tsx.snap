// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> should render like snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root sc-AxiKw jWiChx MuiIconButton-colorInherit MuiIconButton-edgeStart"
        data-testid="IconButtonStyled"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="open drawer"
      class="MuiButtonBase-root MuiIconButton-root sc-AxiKw jWiChx MuiIconButton-colorInherit MuiIconButton-edgeStart"
      data-testid="IconButtonStyled"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
